python/cudf/cudf/core/column/string.py

```diff
@@ -5567,6 +5565,24 @@ def base_size(self) -> int:
         else:
             return self.base_children[0].size - 1
 
+    # override for string column
+    @property
+    def data(self):
+        if self.base_data is None:
+            return None
+        if self._data is None:
+            if (
+                self.offset == 0
+                and len(self.base_children) > 0
+                and self.size == self.base_children[0].size - 1
```

Comment:
**question**: can it be the case that `base_data is not None`  but yet `len(base_children) == 0`? That seems like an impossible case (because that would mean we have a strings column with char data, but no offset data).

Reply:
That case seems impossible. Even if it happens, `self._data = self.base_data[0:0]` will be executed. so, it won't cause an issue.